docs/modules/commons/pages/table-transformers.adoc

```diff
@@ -516,3 +516,96 @@ Examples:
 |value |1    |
 |value2|0    |
 ----
+
+== INNER_JOIN
+
+`INNER_JOIN` transformer combines rows from two tables whenever there are matching values between the columns.
+
+[cols="1,3", options="header"]
+|===
+|Parameter
+|Description
+
+|`leftTableJoinColumn`
+|the column name for matching in the first (left) table
+
+|`rightTableJoinColumn`
+|the column name for matching in the second (right) table
+
+|`tables`
+|paths to examples tables or transformers to inner join
+|===
+
+NOTE: The table is called `left`, when it specified using the first path to xref:ROOT:glossary.adoc#_examplestable[ExamplesTable] or nested transformer in `tables` parameter. Otherwise the table is called `right` (the second path, the second nested transformer or specified as body under transformer)
+
+[IMPORTANT]
+====
+* number of unique tables paths (nested transformers) must be equal 1 if one table is specified as body under transformer, otherwise it must be equal 2
+* the column names of input tables must be different (except the column names for matching)
+* if any of tables is empty - the resulting table will also be empty
+====
+
+.Usage of INNER_JOIN transformer with table body
+[source,gherkin]
+----
+{transformer=INNER_JOIN, leftTableJoinColumn=joinID, rightTableJoinColumn=joinID, tables=/data/some-table.table}
+|joinID|column2|
+|5     |row25  |
+|3     |row23  |
+|1     |row21  |
+----
+
+where xref:ROOT:glossary.adoc#_examplestable[ExamplesTable] from /data/some-table.table:
+
+[source,gherkin]
+----
+|joinID|column1|
+|1     |row11  |
+|2     |row12  |
+|3     |row13  |
+|4     |row14  |
+----
+
+will result in the following xref:ROOT:glossary.adoc#_examplestable[ExamplesTable]:
+
+[source,gherkin]
+----
+|column1|joinID|column2|
+|row11  |1     |row21  |
+|row13  |3     |row23  |
+----
+
+.Usage of INNER_JOIN transformer with the same column names for matching
+[source,gherkin]
+----
+{transformer=INNER_JOIN, leftTableJoinColumn=joinID, rightTableJoinColumn=joinID, tables=story/tables/test1.table;story/tables/test2.table}
+----
+
+.Usage of INNER_JOIN transformer with the different column names for matching
+[source,gherkin]
+----
+{transformer=INNER_JOIN, leftTableJoinColumn=joinID1, rightTableJoinColumn=joinID2, tables=story/tables/test1.table;story/tables/test2.table}
+----
+
+.Usage of INNER_JOIN transformer with nested tables
+[source,gherkin]
+----
+{
+ transformer=INNER_JOIN,
+ leftTableJoinColumn=joinID,
+ rightTableJoinColumn=joinID,
```

Comment:
there are no such columns in tables specified at `tables` parameter

Reply:
updated